samples/matter/common/src/diagnostic/diagnostic_logs_provider.cpp

```diff
@@ -119,19 +122,47 @@ CHIP_ERROR DiagnosticLogProvider::CollectLog(LogSessionHandle sessionHandle, Mut
 		/* TODO: add support for Network Diag Intent */
 		return CHIP_ERROR_NOT_IMPLEMENTED;
 	case IntentEnum::kCrashLogs:
+#ifdef CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_CRASH_LOGS
 		err = GetCrashLogs(outBuffer, outIsEndOfLog);
-		VerifyOrReturnError(CHIP_NO_ERROR == err, err, outBuffer.reduce_size(0));
+		VerifyOrExit(CHIP_NO_ERROR == err, );
 		break;
+#else
+		return CHIP_ERROR_NOT_IMPLEMENTED;
```

Comment:
You don't `outBuffer.reduce_size` in this case. But, in general, why the caller of this function cares about `outBuffer` if it errors out?

Reply:
According to the current API of the `CollectLog` method: https://github.com/ArekBalysNordic/sdk-connectedhomeip/blob/nordic_matter_TE_1_3/src/app/clusters/diagnostic-logs-server/DiagnosticLogsProviderDelegate.h#L68 The caller should check the error code and take care about outBuffer in case of failure. I realized that we don't need to reduce the buffer here.